### ISSUE TITLE
AllVisibleDeck sets --tile-count incorrectly when updating card number

### ISSUE THREAD

**Issue by jordijansen** [thoun/bga-cards#6]
When first creating an AllVisibleDeck and leaving it empty the --tile-count is set to 1 even though no cards are added to it. This is done through CSS:
```
.card-stock.all-visible-deck {
  position: absolute;
  --tile-count: 1;
  width: calc(var(--width) + (var(--tile-count) - 1) * var(--horizontal-shift));
  height: calc(var(--height) + (var(--tile-count) - 1) * var(--vertical-shift));
  transition: width 0.5s ease-in-out, height 0.5s ease-in-out, background-color 0.5s ease-in-out, box-shadow 0.5s ease-in-out;
  opacity: var(--tile-count);
}
```

However, when you add something to it and remove it afterwards, the deck becomes empty. --tile-count is set to 0. It seems like the Stock can't handle having --tile-count set to 0, as it completely disappears. 

I've worked around this issue by overriding the cardNumberUpdated() function like following:
```
class CustomAllVisibleDeck extends AllVisibleDeck<any> {

    protected cardNumberUpdated() {
        const cardNumber = this.cards.length;

        this.element.style.setProperty('--tile-count', ''+ (cardNumber === 0 ? 1 : cardNumber));
        this.element.dataset.empty = (cardNumber == 0).toString();

        const counterDiv = this.element.querySelector('.bga-cards_deck-counter');
        if (counterDiv) {
            counterDiv.innerHTML = `${cardNumber}`;
        }
    }
}
```
Relevant line:
```
        this.element.style.setProperty('--tile-count', ''+ (cardNumber === 0 ? 1 : cardNumber));
```

I'll be happy to open a pull request if this is indeed a bug and should be fixed like I proposed.

**Comment by jordijansen** [comment]
In further inspection:
```
opacity: var(--tile-count);
```

This probably causes the stock to be completely hidden when --tile-count is set to 0.

**Comment by thoun** [comment]
fixed in v3.2.1
